gcloud-java-storage/src/test/java/com/google/gcloud/storage/RemoteGcsHelperTest.java

```diff
@@ -0,0 +1,156 @@
+      + "HBhhhp6mCcRpdQjV38H7JsyJ7lih/oNjECgYAt\\nkndj5uNl5SiuVxHFhcZJO+XWf6ofLUregtevZakGMn8EE1uVa"
+      + "2AY7eafmoU/nZPT\\n00YB0TBATdCbn/nBSuKDESkhSg9s2GEKQZG5hBmL5uCMfo09z3SfxZIhJdlerreP\\nJ7gSi"
+      + "dI12N+EZxYd4xIJh/HFDgp7RRO87f+WJkofMQKBgGTnClK1VMaCRbJZPriw\\nEfeFCoOX75MxKwXs6xgrw4W//AYG"
+      + "GUjDt83lD6AZP6tws7gJ2IwY/qP7+lyhjEqN\\nHtfPZRGFkGZsdaksdlaksd323423d+15/UvrlRSFPNj1tWQmNKk"
+      + "XyRDW4IG1Oa2p\\nrALStNBx5Y9t0/LQnFI4w3aG\\n-----END PRIVATE KEY-----\\n\",\n"
+      + "  \"client_email\": \"someclientid@developer.gserviceaccount.com\",\n"
+      + "  \"client_id\": \"someclientid.apps.googleusercontent.com\",\n"
+      + "  \"type\": \"service_account\"\n"
+      + "}";
+  private static final InputStream JSON_KEY_STREAM = new ByteArrayInputStream(JSON_KEY.getBytes());
+  private static final List<BlobInfo> BLOB_LIST = ImmutableList.of(
+      BlobInfo.builder(BUCKET_NAME, "n1").build(),
+      BlobInfo.builder(BUCKET_NAME, "n2").build());
+  private static final ListResult<BlobInfo> BLOB_LIST_RESULT = new ListResult<BlobInfo>() {
+
+    @Override
+    public String nextPageCursor() {
+      return "listResult";
+    }
+
+    @Override
+    public ListResult<BlobInfo> nextPage() {
+      return null;
+    }
+
+    @Override
+    public Iterator<BlobInfo> iterator() {
+      return BLOB_LIST.iterator();
+    }
+  };
+  private static String KEY_PATH = "/does/not/exist/key." + UUID.randomUUID().toString() + ".json";
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @BeforeClass
+  public static void beforeClass() {
+    while (Files.exists(Paths.get(JSON_KEY))) {
+      KEY_PATH = "/does/not/exist/key." + UUID.randomUUID().toString() + ".json";
+    }
+  }
+
+  @Test
+  public void testForceDelete() throws InterruptedException, ExecutionException {
+    Storage storageMock = EasyMock.createMock(Storage.class);
+    EasyMock.expect(storageMock.list(BUCKET_NAME)).andReturn(BLOB_LIST_RESULT);
+    for (BlobInfo info : BLOB_LIST) {
+      EasyMock.expect(storageMock.delete(BUCKET_NAME, info.name())).andReturn(true);
+    }
+    EasyMock.expect(storageMock.delete(BUCKET_NAME)).andReturn(true);
+    EasyMock.replay(storageMock);
+    assertTrue(RemoteGcsHelper.forceDelete(storageMock, BUCKET_NAME, 5, TimeUnit.SECONDS));
+    EasyMock.verify(storageMock);
+  }
+
+  @Test
+  public void testForceDeleteTimeout() throws InterruptedException, ExecutionException {
+    Storage storageMock = EasyMock.createMock(Storage.class);
+    EasyMock.expect(storageMock.list(BUCKET_NAME)).andReturn(BLOB_LIST_RESULT);
+    for (BlobInfo info : BLOB_LIST) {
+      EasyMock.expect(storageMock.delete(BUCKET_NAME, info.name())).andReturn(true);
+    }
+    EasyMock.expect(storageMock.delete(BUCKET_NAME)).andReturn(false);
+    EasyMock.replay(storageMock);
+    assertTrue(!RemoteGcsHelper.forceDelete(storageMock, BUCKET_NAME, 50, TimeUnit.MICROSECONDS));
```

Comment:
My bad. Unfortunately it is always working locally so I did not realize how bad this test was.